onnxruntime/python/tools/transformers/models/phi2/convert_to_onnx.py

```diff
@@ -0,0 +1,613 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from __future__ import annotations
+
+import argparse
+import logging
+import os
+from enum import Enum
+
+import onnx
+import torch
+from benchmark_helper import Precision
+from dynamo_onnx_helper import DynamoOnnxHelper
+from onnx import ModelProto, TensorProto, helper
+from transformers import AutoConfig, AutoModelForCausalLM
+
+from onnxruntime.quantization.matmul_4bits_quantizer import MatMul4BitsQuantizer
+
+
+class AttentionOpType(Enum):
+    Attention = "Attention"
+    MultiHeadAttention = "MultiHeadAttention"
+    GroupQueryAttention = "GroupQueryAttention"
+
+    def __str__(self):
+        return self.value
+
+
+def env_reset():
+    for flag in ["ATTENTIONOPTYPE"]:
+        if flag in os.environ:
+            del os.environ[flag]
+
+
+class ConvertPhi2ToONNX(DynamoOnnxHelper):
+    def __init__(
+        self,
+        device: torch.device,
+        model_class: str = "microsoft/phi-2",
+        cache_dir: str = "./cache",
+    ):
+        self.model_class = model_class
+        self.device = device
+        self.cache_dir = cache_dir
+        self.phi_config = AutoConfig.from_pretrained(self.model_class, trust_remote_code=True, cache_dir=self.cache_dir)
+        self.phi_model = None
+        self.batch_size = 2
+        self.sequence_length = 8
+        self.phi2_edge_dict = self.get_phi2_edge_dict(self.phi_config)
+        self.attn_op_type = None
+        self.precision = None
+        self.block_size = 16
+        self.accuracy_level = None
+
+    def set_quantization_params(self, block_size: int, accuracy_level: int | None):
+        self.block_size = block_size
+        self.accuracy_level = accuracy_level
+
+    def init_attn_type_and_precision(self, attn_op_type: AttentionOpType, precision: Precision):
+        self.attn_op_type = attn_op_type
+        self.precision = precision
+
+        env_reset()
+        os.environ["ATTENTIONOPTYPE"] = str(attn_op_type)
```

Comment:
What's the usage of the environment variable? I think you can pass the attn_op_type through convert_to_onnx. Once you  have onnx model, you can check onnx nodes to find out the information.

Reply:
The plan is to remove env variables. 
Current design does not allow to use optimizer.py on the fly since a pre-processing step is added in particular for the dynamo exported model. It's possible to move it as post-processing after fusion is done.

Reply:
removed and add attn type to fusion script
code is refactored with decoupling the dynamo export and onnx model preprocessing